presto-main/src/test/java/com/facebook/presto/sql/analyzer/TestMaterializedViewCandidateExtractor.java

```diff
@@ -0,0 +1,219 @@
+        QualifiedName materializedView = QualifiedName.of("materialized_view");
+
+        Map<QualifiedName, List<QualifiedName>> baseTableToMaterializedViews = new HashMap<>();
+        baseTableToMaterializedViews.put(baseTable, ImmutableList.of(materializedView));
+
+        String baseQuerySql = format("SELECT x, y From %s", baseTable);
+        Query baseQuery = (Query) sqlParser.createStatement(baseQuerySql);
+
+        MaterializedViewCandidateExtractor materializedViewCandidateExtractor = new MaterializedViewCandidateExtractor(baseTableToMaterializedViews);
+        materializedViewCandidateExtractor.process(baseQuery);
+        Optional<QualifiedName> materializedViewCandidate = materializedViewCandidateExtractor.getMaterializedViewCandidate();
+
+        checkState(materializedViewCandidate.isPresent(), "Materialized View candidate is not presented");
+
+        assertEquals(materializedView, materializedViewCandidate.get());
+    }
+
+    @Test
+    public void testMaterializedViewCandidateExtractorWithAliasedRelation()
+    {
+        SqlParser sqlParser = new SqlParser();
+
+        QualifiedName baseTable = QualifiedName.of("base_table");
+        QualifiedName materializedView = QualifiedName.of("materialized_view");
+
+        Map<QualifiedName, List<QualifiedName>> baseTableToMaterializedViews = new HashMap<>();
+        baseTableToMaterializedViews.put(baseTable, ImmutableList.of(materializedView));
+
+        String baseQuerySql = format("SELECT x, y From %s b1", baseTable);
+        Query baseQuery = (Query) sqlParser.createStatement(baseQuerySql);
+
+        MaterializedViewCandidateExtractor materializedViewCandidateExtractor = new MaterializedViewCandidateExtractor(baseTableToMaterializedViews);
+        materializedViewCandidateExtractor.process(baseQuery);
+        Optional<QualifiedName> materializedViewCandidate = materializedViewCandidateExtractor.getMaterializedViewCandidate();
+
+        checkState(materializedViewCandidate.isPresent(), "Materialized View candidate is not presented");
+
+        assertEquals(materializedView, materializedViewCandidate.get());
+    }
+
+    @Test
+    public void testMaterializedViewCandidateExtractorWithJoin()
+    {
+        SqlParser sqlParser = new SqlParser();
+
+        QualifiedName baseTable1 = QualifiedName.of("base_table1");
+        QualifiedName baseTable2 = QualifiedName.of("base_table2");
+        QualifiedName materializedView = QualifiedName.of("materialized_view");
+
+        Map<QualifiedName, List<QualifiedName>> baseTableToMaterializedViews = new HashMap<>();
+        baseTableToMaterializedViews.put(baseTable1, ImmutableList.of(materializedView));
+        baseTableToMaterializedViews.put(baseTable2, ImmutableList.of(materializedView));
+
+        String baseQuerySql = format("SELECT x, y FROM %s b1 JOIN %s b2 ON b1.id = b2.id", baseTable1, baseTable2);
+        Query baseQuery = (Query) sqlParser.createStatement(baseQuerySql);
```

Comment:
May be we can create a common method: `assertCandidateMaterializedView`

        Query baseQuery = (Query) sqlParser.createStatement(baseQuerySql);

        MaterializedViewCandidateExtractor materializedViewCandidateExtractor = new MaterializedViewCandidateExtractor(baseTableToMaterializedViews);
        materializedViewCandidateExtractor.process(baseQuery);
        Optional<QualifiedName> materializedViewCandidate = materializedViewCandidateExtractor.getMaterializedViewCandidate();

        checkState(materializedViewCandidate.isPresent(), "Materialized View candidate is not presented");

        assertEquals(materializedView, materializedViewCandidate.get());